### PR TITLE
feat: Added number_of_ variables for iam-assumable-role submodules

### PR DESCRIPTION
**Pull request by Zyntogz**
fix for terraform error for assumable-role"number of resources cannot be determined during runtime" in iam-assumable-role

### CODE REVIEW

File: modules/iam-assumable-role-with-oidc/main.tf [hunk 1]
```diff
@@ -66,7 +66,7 @@ resource "aws_iam_role" "this" {
 }
 
 resource "aws_iam_role_policy_attachment" "custom" {
-  count = var.create_role ? length(var.role_policy_arns) : 0
+  count = var.create_role ? var.number_role_policy_arns : 0
```
Comment (antonbabenko) [inline modules/iam-assumable-role-with-oidc/main.tf:69]: `number_of_role_policy_arns`

File: modules/iam-assumable-role/variables.tf [hunk 2]
```diff
@@ -76,6 +76,12 @@ variable "custom_role_policy_arns" {
   default     = []
 }
 
+variable "number_custom_role_policy_arns" {
```
Comment (antonbabenko) [inline modules/iam-assumable-role/variables.tf:79]: `number_of_custom_role_policy_arns`

**Review by antonbabenko — COMMENTED** [review]
Good. Also, please update examples and docs (run `pre-commit run -a`), if you can.

**Comment by antonbabenko** [comment]
Thank you!

[v2.26.0](https://github.com/terraform-aws-modules/terraform-aws-iam/releases/v2.26.0) has been just released.

**Comment by llamahunter** [comment]
Please revert this change. It is not backwards compatible.

**Comment by antonbabenko** [comment]
[v3.1.0](https://github.com/terraform-aws-modules/terraform-aws-iam/releases/v3.1.0) has been just released with this PR, while [v2.25.0](https://github.com/terraform-aws-modules/terraform-aws-iam/releases/v2.25.0) is the latest in v2 where the behavior is the same as before this PR was merged.

Apologize for the confusion.

**Comment by horjulf** [comment]
Wouldn't leveraging `for_each` be a better approach here ?
This is very error prone, especially with the default `0`.

**Comment by antonbabenko** [comment]
@horjulf `for_each` will act exactly like `count` in this case. There is no better workaround in Terraform 0.13 for computed values being used in meta-arguments like `count` and `for_each`.

**Pull request merged**